cli/cmd/deploy/deploy.go

```diff
@@ -305,17 +383,25 @@ func extractRemote(remotePath string) (string, error) {
 	return gitutil.RemotesToGithubURL(remotes)
 }
 
-func exitWithFailure(err error) {
-	errormsg := color.New(color.Bold).Add(color.FgRed)
-	errormsg.Printf("Prompt failed %v\n", err)
-	os.Exit(1)
-}
-
 func hasRillProject(dir string) bool {
 	_, err := os.Open(filepath.Join(dir, "rill.yaml"))
 	return err == nil
 }
 
+func projectName(dir string) (string, error) {
+	content, err := os.ReadFile(filepath.Join(dir, "rill.yaml"))
+	if err != nil {
+		return "", err
+	}
+
+	c, err := rillv1beta.ParseProjectConfig(content)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.ReplaceAll(strings.TrimSpace(c.Name), " ", "-"), nil
+}
```

Comment:
1. Move to `rillv1beta`
2. Probably best to replace all non-alphanumeric characters with `-`. Maybe also replace `-+` with `-`. And if it's all dashes, prompt for name instead. (Test case: `name: 💣💣💣`)